modules/offload-s3/module.go

```diff
@@ -294,16 +302,43 @@ func (m *Module) Download(ctx context.Context, className, shardName, nodeName st
 
 // Delete deletes content of a shard assigned to specific node in
 // cloud provider (S3, Azure Blob storage, Google cloud storage)
+// Careful: if shardName and nodeName is passed empty it will delete all class frozen shards in cloud storage
 // {cloud_provider}://{configured_bucket}/{className}/{shardName}/{nodeName}/{shard content}
 func (m *Module) Delete(ctx context.Context, className, shardName, nodeName string) error {
+	if className == "" {
+		return fmt.Errorf("can't pass empty class name")
+	}
+
+	if shardName == "" && nodeName != "" {
+		return fmt.Errorf("can't pass empty shard name")
+	}
+
+	if nodeName == "" && shardName != "" {
+		return fmt.Errorf("can't pass empty node name")
+	}
+
 	ctx, cancel := context.WithTimeout(ctx, m.timeout)
 	defer cancel()
+
+	cloudPath := fmt.Sprintf("s3://%s/%s/%s/%s/*", m.Bucket, strings.ToLower(className), shardName, nodeName)
+
+	if shardName == "" && nodeName == "" {
```

Comment:
hmm doesn't previous checks (line 312 and line 316) ensure enough that we here always will have not empty shard and node names?

Reply:
here actually it's not validation , it's updating the path to make sure to delete all the shards under a class